### ISSUE TITLE
Sorting no longer working after updating to latest version and Forge 28.1.0

### ISSUE THREAD

**Issue by apollo-mg** [cpw/inventorysorter#78]
**Edit 2: Inventory sorting started working after opening a chest, but only while the chest is open**

Updated Sorter from 15.2.0 to 16.0.0, Forge from 28.0.83 to 28.1.0. Sorting no longer works.

Edit: Also updated Optifine to OptiFine 1.14.4 HD U F4 pre4, previously pre2.

Edit 2: Inventory sorting started working after opening a chest.

**Comment by Elsker99** [comment]
This issue is occurring for me too. Any container being opened is the only way the sorter works. just wanted to give some feedback that it's not just 1 person dealing with this

**Comment by Ackmard** [comment]
Same issue, sorting only works while viewing any inventory other than just your own, like having a chest open or looking at furnace

**Comment by CapeyBoy** [comment]
I have tried to use this mod as InventoryTweaks isn't available for 1.14.4 but this mod hasn't worked at all for me. Pressing the middle mouse button (Scroll Wheel) it has never sorted my items. How do I fix this?

**Comment by WenXin20** [comment]
I'm using Forge v28.1.1
Here is what's printed in the log when sorting is attempted:
```
[01:32:29] [Server thread/FATAL]: Error executing task on Server
java.lang.UnsupportedOperationException: Unable to construct this menu by type
	at net.minecraft.inventory.container.Container.func_216957_a(Container.java:52) ~[?:?]
	at cpw.mods.inventorysorter.SortingHandler.compactInventory(SortingHandler.java:114) ~[?:16.0.0]
	at cpw.mods.inventorysorter.SortingHandler.accept(SortingHandler.java:52) ~[?:16.0.0]
	at cpw.mods.inventorysorter.SortingHandler.accept(SortingHandler.java:35) ~[?:16.0.0]
	at cpw.mods.inventorysorter.Action.execute(Action.java:68) ~[?:16.0.0]
	at cpw.mods.inventorysorter.ServerHandler.lambda$onMessage$0(ServerHandler.java:38) ~[?:16.0.0]
	at cpw.mods.inventorysorter.ServerHandler$$Lambda$10869/1048221438.run(Unknown Source) ~[?:?]
	at net.minecraftforge.fml.network.NetworkEvent$Context.enqueueWork(NetworkEvent.java:185) ~[?:?]
	at cpw.mods.inventorysorter.ServerHandler.onMessage(ServerHandler.java:35) ~[?:16.0.0]
	at cpw.mods.inventorysorter.Network$$Lambda$6188/1536233926.accept(Unknown Source) ~[?:?]
	at net.minecraftforge.fml.network.simple.IndexedMessageCodec.lambda$tryDecode$3(IndexedMessageCodec.java:114) ~[?:?]
	at net.minecraftforge.fml.network.simple.IndexedMessageCodec$$Lambda$10054/1537733528.accept(Unknown Source) ~[?:?]
	at java.util.Optional.ifPresent(Optional.java:159) ~[?:1.8.0_51]
	at net.minecraftforge.fml.network.simple.IndexedMessageCodec.tryDecode(IndexedMessageCodec.java:114) ~[?:?]
	at net.minecraftforge.fml.network.simple.IndexedMessageCodec.consume(IndexedMessageCodec.java:147) ~[?:?]
	at net.minecraftforge.fml.network.simple.SimpleChannel.networkEventListener(SimpleChannel.java:65) ~[?:?]
	at net.minecraftforge.fml.network.simple.SimpleChannel$$Lambda$1936/791091938.accept(Unknown Source) ~[?:?]
	at net.minecraftforge.eventbus.EventBus.doCastFilter(EventBus.java:212) ~[eventbus-1.0.0-service.jar:?]
	at net.minecraftforge.eventbus.EventBus.lambda$addListener$11(EventBus.java:204) ~[eventbus-1.0.0-service.jar:?]
	at net.minecraftforge.eventbus.EventBus$$Lambda$1938/1481923630.invoke(Unknown Source) ~[?:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:258) ~[eventbus-1.0.0-service.jar:?]
	at net.minecraftforge.fml.network.NetworkInstance.dispatch(NetworkInstance.java:82) ~[?:?]
	at net.minecraftforge.fml.network.NetworkHooks.lambda$onCustomPayload$0(NetworkHooks.java:69) ~[?:?]
	at net.minecraftforge.fml.network.NetworkHooks$$Lambda$10042/1653507553.apply(Unknown Source) ~[?:?]
	at java.util.Optional.map(Optional.java:215) ~[?:1.8.0_51]
	at net.minecraftforge.fml.network.NetworkHooks.onCustomPayload(NetworkHooks.java:69) ~[?:?]
	at net.minecraft.network.play.ServerPlayNetHandler.func_147349_a(ServerPlayNetHandler.java:1275) ~[?:?]
	at net.minecraft.network.play.client.CCustomPayloadPacket.func_148833_a(CCustomPayloadPacket.java:42) ~[?:?]
	at net.minecraft.network.play.client.CCustomPayloadPacket.func_148833_a(CCustomPayloadPacket.java:12) ~[?:?]
	at net.minecraft.network.PacketThreadUtil.func_225383_a(SourceFile:21) ~[?:?]
	at net.minecraft.network.PacketThreadUtil$$Lambda$10143/1802467117.run(Unknown Source) ~[?:?]
	at net.minecraft.util.concurrent.TickDelayedTask.run(SourceFile:18) ~[?:?]
	at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213166_h(SourceFile:144) [?:?]
	at net.minecraft.util.concurrent.RecursiveEventLoop.func_213166_h(SourceFile:23) [?:?]
	at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213168_p(SourceFile:118) [?:?]
	at net.minecraft.server.MinecraftServer.func_213205_aW(MinecraftServer.java:699) [?:?]
	at net.minecraft.server.MinecraftServer.func_213168_p(MinecraftServer.java:693) [?:?]
	at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213161_c(SourceFile:127) [?:?]
	at net.minecraft.server.MinecraftServer.func_213202_o(MinecraftServer.java:679) [?:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:626) [?:?]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_51]
[01:32:34] [Client thread/ERROR]: ########## GL ERROR ##########
[01:32:34] [Client thread/ERROR]: @ Pre render
[01:32:34] [Client thread/ERROR]: 65539: Invalid key 0
```

**Comment by skweezu** [comment]
I'm also having this same issue. I thought at first it might be a conflict with Cosmetic Armor Reworked, but it persists after removing that mod. I don't have anything else that messes with the inventory screen aside from JEI, but it was previously working fine with JEI. Using Forge version 28.1.26 on Minecraft version 1.14.4.

**Comment by fuzzyweapon** [comment]
Just for a little information on why it might be more important to fix this bug.  A lot of mods are requiring later Forge versions.  I had to go to 28.1.10, for instance, and can no longer target the optimal forge version for this mod, but it /never/ worked for me on 1.14.4, even on Forge 28.0.x.

I think it's possible that the 1.14.4 Release was more like an alpha release, where crashes were fixed but functionality was not re-verified
